src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/OpenFolderDialog.cs

```
@@ -0,0 +1,645 @@
+        public bool AddToMruList { get; set; } = true;
+
+        ///<summary>
+        /// Gets or sets a value that controls whether to show hidden items.
+        /// </summary>
+        public bool ShowHiddenItems { get; set; }
+
+        ///<summary>
+        /// Gets or sets a value that controls whether shortcuts should be treated as their target items, allowing an application to open a .lnk file.
+        /// Default value is true.
+        /// </summary>
+        public bool NavigateToShortcut { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets a value that determines whether the user can select non-filesystem items, 
+        /// such as <b>Library</b>, <b>Search Connectors</b>, or <b>Known Folders</b>.
+        /// </summary>
+		public bool AllowNonFileSystem { get; set; }
+
+        /// <summary>
+        /// Gets or sets the initial directory displayed when the dialog is shown. A null or empty string indicates that the dialog is using the default directory.
+        /// </summary>
+        public string InitialDirectory { get; set; }
+        #endregion
+
+        //---------------------------------------------------
+        //
+        // Private Methods
+        //
+        //---------------------------------------------------
+        #region Private Methods
+        private void ApplyNativeSettings(IFileDialog dialog)
+        {
+            dialog.SetTitle(Title);
+            NativeMethods.FOS flags = ApplyNativeDialogFlags();
+            dialog.SetOptions(flags);
+
+            string directory = string.IsNullOrEmpty(InitialDirectory) ? null : Path.GetDirectoryName(InitialDirectory);
+            if (directory == null) return;
+            try
+            {
+                NativeMethods.SHCreateItemFromParsingName(InitialDirectory, IntPtr.Zero, new Guid(IIDGuid.IShellItem), out IShellItem folder);
+                if (folder != null)
+                    dialog.SetFolder(folder);
+            }
+            catch
```

Should we catch all exception?